### ISSUE TITLE
Passing configuration properties as command line arguments aren't found

### ISSUE THREAD

**Issue by kkroner8451** [paulschwarz/spring-dotenv#12]
I'm trying to pass the `.env.directory` as a command line argument and it is not resolved in the `DotenvConfig` class. I've tried to run under Maven such as :
```
$ mvn spring-boot:run -Drun.arguments=--.env.directory=/fully/qualified/path/env
```
or directly from executable jar:
```
$ java -D.env.directory=/fully/qualified/path/env -jar my.jar
```

I've passed other command line arguments in and while debugging I can see they are resolved using `environment.getProperty(key)` at the time `DotenvConfig` is instantiated. I've also been able to set an environment variable of `.env.directory` that as long as it is sourced before running the command line works. It seems that Java, SpringBoot or other (native system?) doesn't like command like argument names starting with dot (".") and doesn't work as expected. I'm not sure if it treats them as hidden or just doesn't parse them correctly but the net result is they aren't available. 

Do you see anything I'm doing wrong or would you consider changing the CONFIG_PREFIX to something that doesn't start with a dot or offer a fallback lookup prefix to support this use case?

**Comment by paulschwarz** [comment]
Hi @kkroner8451 I appreciate you reporting this and I had some time to look into this today. 

As you can see, [spring-dotenv](https://github.com/paulschwarz/spring-dotenv) is just a Spring property source on top of [dotenv-java](https://github.com/cdimascio/dotenv-java) and I am keen to keep this transparent without too much transformation before delegating to the underlying library.

Having said that, when referring to the underlying library I noticed something curious. Notice here https://github.com/cdimascio/dotenv-java#advanced-usage how the path begins with `./`. (And then later under https://github.com/cdimascio/dotenv-java#optional-directory the example is without the `./`).

In short, when I places my **.env** file in **src/main/resources/some/path/.env** this worked for:

`-D.env.directory=./some/path`

... while this did not (without the `./`):

`-D.env.directory=/some/path`

You can see this more clearly with:

`-D.env.directory=/some/path -D.env.ignoreIfMissing=false`

I wonder if @cdimascio can offer an explanation when/why one would use the `/path` vs `./path`?

**Comment by paulschwarz** [comment]
Coincidentally, I just saw a [PR](https://github.com/cdimascio/dotenv-kotlin/pull/65/files) on the Kotlin version of upstream library which seems to indicate the same issue. @cdimascio that PR should probably be repeated on the Java version too.

**Comment by paulschwarz** [comment]
Updated docs. Just remember to start your directory with "./"
